hermes-common/src/main/java/pl/allegro/tech/hermes/common/config/Configs.java

```diff
@@ -104,8 +116,12 @@
     MESSAGES_LOADING_WAIT_FOR_TOPICS_CACHE("frontend.messages.loading.wait.for.topics.cache", 10),
     MESSAGES_LOADING_WAIT_FOR_BROKER_TOPIC_INFO("frontend.messages.loading.wait.for.broker.topic.info", 5),
 
-    CONSUMER_COMMIT_OFFSET_PERIOD("consumer.commit.offset.period", 15),
-    CONSUMER_COMMIT_OFFSET_QUEUES_SIZE("consumer.commit.offset.queues.size", 500_000),
+    CONSUMER_RECEIVER_POOL_TIMEOUT("consumer.receiver.pool.timeout", 100),
+    CONSUMER_RECEIVER_READ_QUEUE_CAPACITY("consumer.receiver.read.queue.capacity", 1000),
+
+    CONSUMER_COMMIT_OFFSET_PERIOD("consumer.commit.offset.period.ms", 15 * 1000),
+    CONSUMER_COMMIT_OFFSET_QUEUES_SIZE("consumer.commit.offset.queues.size", 1000),
```

Comment:
Good question. Flattening of offsets is done by offset committer each N seconds (15 now?). We could flatten this structure earlier, but anyway we have the worst case scenario of offsets that can't be arranged into ranges. And running flattening is expensive (and x 300 subscriptions might become time consuming). What is worse, offsets are managed from different places in code, different threads as well. So it might not be so easy. Thus the queue: just fire and forget (quick) and let some process consume it later.

Now what we need is a queue in size which will be allow it to wait N seconds under heavy load (1k? 2k? 3k? rps) before it gets flattened.